File: android/app/src/main/java/com/hardwarestore/manager/data/model/Transaction.kt
package com.hardwarestore.manager.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

enum class TransactionType {
    SALE,       // Daily Sale
    RETURN,     // Customer Goods Return (Motiry Return)
    PAYMENT     // Customer Udhar / Credit Repayment
}

enum class PaymentMethod {
    CASH,
    CREDIT      // Borrow / Udhar
}

@Entity(tableName = "transactions")
data class Transaction(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val customerId: Long? = null,
    val customerName: String = "",
    val customerPhone: String = "",
    val type: TransactionType,
    val paymentMethod: PaymentMethod,
    val totalAmount: Double,
    val notes: String = "",
    val timestamp: Long = System.currentTimeMillis(),
    val dateString: String // YYYY-MM-DD for date-based daily indexing
)

File: server/models/Transaction.js
import mongoose from 'mongoose'

const itemSchema = new mongoose.Schema({
  name: { type: String, required: true },
  qty: { type: Number, required: true, default: 1 },
  unitPrice: { type: Number, required: true, default: 0 },
  total: { type: Number, required: true, default: 0 }
}, { _id: false })

const transactionSchema = new mongoose.Schema({
  type: {
    type: String,
    enum: ['SALE', 'RETURN', 'PAYMENT'],
    required: true
  },
  paymentMethod: {
    type: String,
    enum: ['CASH', 'CREDIT'],
    required: true
  },
  customerName: { type: String, default: '' },
  customerPhone: { type: String, default: '' },
  customerId: { type: mongoose.Schema.Types.ObjectId, ref: 'Customer', default: null },
  dateString: { type: String, required: true }, // YYYY-MM-DD
  totalAmount: { type: Number, required: true, default: 0 },
  items: [itemSchema],
  notes: { type: String, default: '' },
  createdAt: { type: Date, default: Date.now }
})

export default mongoose.model('Transaction', transactionSchema)
